include/fuse_common.h

```diff
@@ -114,6 +114,9 @@ struct fuse_file_info {
 	 * create and open.  It is used to create a passthrough connection
 	 * between FUSE file and backing file. */
 	int32_t backing_id;
+
+	/* For future additions and ABI compatibility */
+	uint64_t reserved[32];
 };
```

Comment:
```
#define STATIC_ASSERT(COND,MSG) typedef char static_assertion_##MSG[(COND)?1:-1]

STATIC_ASSERT(sizeof(A) == 123, struct_size_must_not_change); 
```

Something like this might be a possibility to avoid struct size changes with something stronger than comments.

Reply:
I won't work for single bits - the issue in `struct fuse_config` was not beautiful, but mostly not critical. Shifted bits in `struct fuse_file_info` had been really bad. I prefer to go with external ABI verification tools, at least that is going to make me to look carefully if it reports something, even if it is a false positive.